lib/common/validation.js

```diff
@@ -908,6 +907,105 @@ function createValidateMarkerFn(field, options) {
     };
 }
 
+function validVolumeName(volumeName) {
+    var VALID_VOLUME_NAME_REGEXP = /^[a-zA-Z0-9][a-zA-Z0-9_\.\-]+$/;
+    if (volumeName === undefined || volumeName === null ||
+        !VALID_VOLUME_NAME_REGEXP.test(volumeName)) {
+        return false;
+    }
+
+    return true;
+}
+
+function validVolumeMountpoint(mountpoint) {
+    if (typeof (mountpoint) !== 'string') {
+        return false;
+    }
+
+    /*
+     * Must start with '/'.
+     */
+    if (mountpoint.length === 0 || mountpoint[0] !== '/') {
+        return false;
+    }
+
+    /*
+     * Must not contain the null character.
+     */
+    if (mountpoint.indexOf('\0') !== -1) {
+        return false;
+    }
+
+    /*
+     * Must contain at least one character that is not '/'.
+     */
+    if (mountpoint.search(/[^\/]/) === -1) {
+        return false;
+    }
+
+    return true;
+}
+
+function createValidateVolumes(fieldName, required) {
+    assert.string(fieldName, 'fieldName');
+    assert.optionalBool(required, 'required');
+
+    return function validateVolumesParam(params) {
+        var errs = [];
+        var idx = 0;
+        var VALID_VOLUME_MODES = ['ro', 'rw'];
+        var volume;
+        var volumes = params[fieldName];
+
+        if (required === false && volumes === undefined) {
+            return errs;
+        }
+
+        if (!Array.isArray(volumes)) {
+            errs.push(errors.invalidParamErr(fieldName,
+                    fieldName + ' must be an array of objects'));
```

Comment:
No, you're right, I don't know how that and other lines of code ended up being over-indented. Will fix, thanks!